src/components/body/ammo-body.js

```diff
@@ -0,0 +1,480 @@
+/* global Ammo,THREE */
+const AmmoDebugDrawer = require("ammo-debug-drawer");
+const threeToAmmo = require("three-to-ammo");
+
+//Activation States
+const ACTIVE_TAG = 1;
+const ISLAND_SLEEPING = 2;
+const WANTS_DEACTIVATION = 3;
+const DISABLE_DEACTIVATION = 4;
+const DISABLE_SIMULATION = 5;
+
+//CollisionFlags
+const CF_STATIC_OBJECT = 1;
+const CF_KINEMATIC_OBJECT = 2;
+const CF_NO_CONTACT_RESPONSE = 4;
+const CF_CUSTOM_MATERIAL_CALLBACK = 8; //this allows per-triangle material (friction/restitution)
+const CF_CHARACTER_OBJECT = 16;
+const CF_DISABLE_VISUALIZE_OBJECT = 32; //disable debug drawing
+const CF_DISABLE_SPU_COLLISION_PROCESSING = 64; //disable parallel/SPU processing
+
+function almostEqualsVector3(epsilon, u, v) {
+  return Math.abs(u.x - v.x) < epsilon && Math.abs(u.y - v.y) < epsilon && Math.abs(u.z - v.z) < epsilon;
+}
+
+function almostEqualsQuaternion(epsilon, u, v) {
+  return (
+    (Math.abs(u.x - v.x) < epsilon &&
+      Math.abs(u.y - v.y) < epsilon &&
+      Math.abs(u.z - v.z) < epsilon &&
+      Math.abs(u.w - v.w) < epsilon) ||
+    (Math.abs(u.x + v.x) < epsilon &&
+      Math.abs(u.y + v.y) < epsilon &&
+      Math.abs(u.z + v.z) < epsilon &&
+      Math.abs(u.w + v.w) < epsilon)
+  );
+}
+
+let AmmoBody = {
+  schema: {
+    loadedEvent: { default: "" },
+    mass: { default: 1 },
+    gravity: { type: "vec3", default: { x: 0, y: -9.8, z: 0 } },
+    linearDamping: { default: 0.01 },
+    angularDamping: { default: 0.01 },
+    linearSleepingThreshold: { default: 1.6 },
+    angularSleepingThreshold: { default: 2.5 },
+    angularFactor: { type: "vec3", default: { x: 1, y: 1, z: 1 } },
+    activationState: {
+      default: ACTIVE_TAG,
+      oneOf: [ACTIVE_TAG, ISLAND_SLEEPING, WANTS_DEACTIVATION, DISABLE_DEACTIVATION, DISABLE_SIMULATION]
+    },
+    type: { default: "dynamic", oneOf: ["static", "dynamic", "kinematic"] },
+    addCollideEventListener: { default: false },
```

Comment:
Why would we not want this? Is there extra cost on AmmoJS? Since the effect from the user's perspective is that there is an event on the entity or there is not, then I think I would name this `emitCollideEvents` or `emitOnCollide`.

Reply:
I agree the name should probably be changed, but we wouldn't want this on by default because calling emit isn't free, afaik.

Reply:
Nothing is free, but it's on by default in CannonJS and I've never measured an impact of that. I could imagine it being costly if there are 100 objects on the floor, and they emit on every frame. Assuming Bullet works similarly to CannonJS and doesn't debounce collisions.

Anyway, I'm fine with whichever default if we rename it.